torch/distributed/fsdp/_symbolic_trace.py

```diff
@@ -0,0 +1,202 @@
+    tracer: An instance of ``torch.fx.Tracer`` that will be used to perform symbolic
+        tracing. ``tracer`` is default to be ``torch.fx.Tracer()``, but can also be instance
+        of some child class of ``torch.fx.Tracer``. For example, one may want to use ``HFTracer``
+        for models in Transformers:
+        .. _Transformers: https://huggingface.co/docs/transformers/index
+
+    concrete_args: Concrete arguments that should not be treated as ``torch.fx.Proxy``
+        when tracing the forward function.
+        ``concrete_args`` allows one to partially specialize the forward function,
+        including removing control flow or data structures.
+        ``concrete_args`` is also the parameter used in ``tracer.trace()``:
+        https://pytorch.org/docs/stable/fx.html#torch.fx.Tracer.trace
+    """
+
+    tracer: torch.fx.Tracer = torch.fx.Tracer()
+    concrete_args: Optional[Dict[str, Any]] = None
+
+
+class _ExecutionInfo:
+    """
+    Contains the execution order information in the model forward pass.
+
+    Attributes:
+        current_module: record the module that is currently being traced.
+
+        module_forward_order: a list of modules, where the ordering is based on
+            when their forward function is called.
+
+        param_exec_order: a list of parameters ordered based on their execution order.
+
+        _traced_param_set: a set containing all parameters that have been traced.
+
+        module_execution_info_dict: a dict that maps each module to a list of
+            tuple containing a module and a list of named parameters.
+            For a given module, each tuple:
+            1. either contains this module and part of its ``named_parameters`` that will be executed together,
+            2. or contains one of its child modules and all of the child module's ``named_parameters``.
+            The list of tuple is ordered based on the parameter execution order.
+    """
+
+    def __init__(self, root_module: torch.nn.Module) -> None:
+        self.current_module = root_module
+        self.module_forward_order: List[torch.nn.Module] = [root_module]
+
+        named_params_type = List[Tuple[str, torch.nn.Parameter]]
+        self.module_execution_info_dict: Dict[
+            torch.nn.Module, List[Tuple[torch.nn.Module, named_params_type]]
+        ] = dict()
+        self.module_execution_info_dict[root_module] = []
+
+        self.param_exec_order: List[torch.nn.Parameter] = []
+        self._traced_param_set: Set[torch.nn.Parameter] = set()
+
+
+def _patched_create_proxy(
+    create_proxy: Callable,
+    execution_info: _ExecutionInfo,
+    params_dict: Dict[str, torch.nn.Parameter],
+    kind: str,
+    target: torch.fx.node.Target,
+    args: Tuple[Any, ...],
+    kwargs: Dict[str, Any],
+    name: Optional[str] = None,
+    type_expr: Optional[Any] = None,
+):
+    """
+    Override of ``Tracer.create_proxy`` (see
+    https://pytorch.org/docs/stable/fx.html#torch.fx.Tracer.create_proxy).
```

Comment:
We usually don't put raw URLs in docs. For method/functions/types in PyTorch, you can try this

https://github.com/pytorch/pytorch/blob/c28315eab851b9d126457738d73deae0cccfc2bc/torch/distributed/rpc/api.py#L317-L318

Reply:
there are another places used raw URLs, please update them too